apps/financial-aid/web-osk/src/routes/status/FileUpload/fileUpload.tsx

```diff
@@ -18,45 +17,42 @@ import { useMutation } from '@apollo/client'
 import { UpdateApplicationMutation } from '@island.is/financial-aid-web/oskgraphql/sharedGql'
 
 const FileUpload = () => {
-  const { form } = useContext(FormContext)
+  const { form, updateForm } = useContext(FormContext)
   const router = useRouter()
-  const [nextButtonText, setNextButtonText] = useState('Senda gögn')
   const { uploadFiles } = useFileUpload(form.otherFiles)
-  const { user } = useContext(UserContext)
-
-  const currentApplication = useMemo(() => {
-    if (user?.currentApplication) {
-      return user.currentApplication
-    }
-  }, [user])
+  const [isLoading, setIsLoading] = useState(false)
 
   const [updateApplicationMutation] = useMutation<{ application: Application }>(
     UpdateApplicationMutation,
   )
 
   const proceed = async () => {
-    if (form?.otherFiles.length <= 0 || currentApplication === undefined) {
-      setNextButtonText('Engar skrár til staðar')
+    if (form?.otherFiles.length <= 0 || router.query.id === undefined) {
       return
     }
 
+    setIsLoading(true)
+
     try {
-      await uploadFiles(currentApplication.id, FileType.OTHER).then(
+      await uploadFiles(router.query.id as string, FileType.OTHER).then(
```

Comment:
as string check is this necessary? 🤓

Reply:
Yes because the type is `string | string[]` 😢